packages/grafana-data/src/transformations/transformDataFrame.ts

```diff
@@ -1,37 +1,68 @@
+import { MonoTypeOperatorFunction, Observable, of, OperatorFunction } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
+
 import { DataFrame, DataTransformerConfig } from '../types';
-import { standardTransformersRegistry } from './standardTransformersRegistry';
+import { standardTransformersRegistry, TransformerRegistyItem } from './standardTransformersRegistry';
 
-/**
- * Apply configured transformations to the input data
- */
-export function transformDataFrame(options: DataTransformerConfig[], data: DataFrame[]): DataFrame[] {
-  let processed = data;
-  for (const config of options) {
-    const info = standardTransformersRegistry.get(config.id);
-
-    if (!info) {
-      return data;
-    }
-
-    const defaultOptions = info.transformation.defaultOptions ?? {};
-    const options = { ...defaultOptions, ...config.options };
-    const transformer = info.transformation.transformer(options);
-    const after = transformer(processed);
-
-    // Add a key to the metadata if the data changed
-    if (after && after !== processed) {
+const getOperator = (config: DataTransformerConfig): MonoTypeOperatorFunction<DataFrame[]> => source => {
+  const info = standardTransformersRegistry.get(config.id);
+
+  if (!info) {
+    return source;
+  }
+
+  const defaultOptions = info.transformation.defaultOptions ?? {};
+  const options = { ...defaultOptions, ...config.options };
+
+  return source.pipe(
+    mergeMap(before => of(before).pipe(info.transformation.operator(options), postProcessTransform(before, info)))
+  );
+};
+
+const postProcessTransform = (
+  before: DataFrame[],
+  info: TransformerRegistyItem<any>
+): MonoTypeOperatorFunction<DataFrame[]> => source =>
```

Comment:
nit: same thing here. You could use the `pipe` utility function instead to clean up the code.

Reply:
It is basically the same thing but for me it removes some noise.

Reply:
Same as above